lib/arjdbc/postgresql/base/array_encoder.rb

```diff
@@ -0,0 +1,21 @@
+# This implements a basic encoder to work around ActiveRecord's dependence on the pg gem
+module PG
+  module TextEncoder
+    class Array
+
+      def initialize(name:, delimiter:)
+        @type = if name == 'string[]'.freeze
```

Comment:
Seems reasonable as a workaround.  My only other thought is perhaps the version of JDBC somehow fixes something?